sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkAsyncHttpClient.java

```diff
@@ -216,14 +216,36 @@ public HttpHeaders getHeaders() {
 
         @Override
         public Mono<String> getBodyAsString() {
-            return getBodyAsByteArray()
-                .map(bytes -> new String(bytes, StandardCharsets.UTF_8));
+            return getBodyAsByteArray().map(bytes -> {
+                if (bytes.length >= 3 && bytes[0] == (byte) 239 && bytes[1] == (byte) 187 && bytes[2] == (byte) 191) {
+                    return new String(bytes, 3, bytes.length - 3, StandardCharsets.UTF_8);
+                } else if (bytes.length >= 2 && bytes[0] == (byte) 254 && bytes[1] == (byte) 255) {
+                    return new String(bytes, 2, bytes.length - 2, StandardCharsets.UTF_16BE);
+                } else if (bytes.length >= 2 && bytes[0] == (byte) 255 && bytes[1] == (byte) 254) {
+                    return new String(bytes, 2, bytes.length - 2, StandardCharsets.UTF_16LE);
```

Comment:
Any reason why we look for only these 3 charsets and not others? Adding a comment here to explain that would be good.

Reply:
Added support for `UTF-32BE` and `UTF-32LE`